Description: use Compilation definition to use or not MIX_INIT_FLUIDSYNTH
Author: Nicolas HOUDELOT <nicolas@demosdebs.org>
Applied-Upstream: no
Last-Update: 2024-04-24
---
This patch header follows DEP-3: http://dep.debian.net/deps/dep3/
--- a/mixer.cpp
+++ b/mixer.cpp
@@ -208,7 +208,12 @@
 		}
 		_sfx = 0;
 
-		Mix_Init(MIX_INIT_OGG | MIX_INIT_FLUIDSYNTH);
+		#ifdef MIX_INIT_FLUIDSYNTH
+			Mix_Init(MIX_INIT_OGG | MIX_INIT_FLUIDSYNTH);
+		#endif
+		#ifdef MIX_INIT_MID
+			Mix_Init(MIX_INIT_OGG | MIX_INIT_MID);
+		#endif
 		if (Mix_OpenAudio(kMixFreq, kMixFormat, kMixSoundChannels, kMixBufSize) < 0) {
 			warning("Mix_OpenAudio failed: %s", Mix_GetError());
 		}
